File: Data Structures & Algorithms/buy-and-sell-crypto/submission-3.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int currProfit = 0;
        int l = 0;
        int r = 1;
        while(r<prices.length) {
            if(prices[l]>prices[r]) {
                l = r;
            }
            else {
                currProfit = prices[r] - prices[l];
                max = Math.max(currProfit, max);
            }
            r++;
        }   
        return max;

    }
}

File: Data Structures & Algorithms/eating-bananas/submission-3.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l = 1           # min k
        r = max(piles)  # max k
        mid = (l+r) // 2
        min_k = r

        while l <= r:
            time = 0
            for pile in piles:
                time += math.ceil(pile/mid)
            if time <= h:
                r = mid - 1
                min_k = min(min_k, mid)
            else:
                l = mid + 1

            mid = (l + r) // 2

        return min_k
        